timetypes/go_duration_value_test.go

```diff
@@ -175,6 +175,64 @@ func TestDurationValidateParameter(t *testing.T) {
 	}
 }
 
+func TestDuration_StringSemanticEquals(t *testing.T) {
+	t.Parallel()
+
+	testCases := map[string]struct {
+		currentDuration timetypes.GoDuration
+		givenDuration   basetypes.StringValuable
+		expectedMatch   bool
+		expectedDiags   diag.Diagnostics
+	}{
+		"not equal - different durations": {
+			currentDuration: timetypes.NewGoDurationValueFromStringMust("50s"),
+			givenDuration:   timetypes.NewGoDurationValueFromStringMust("50m"),
+			expectedMatch:   false,
+		},
+		"equal - exactly the same string": {
+			currentDuration: timetypes.NewGoDurationValueFromStringMust("30h22m33s"),
+			givenDuration:   timetypes.NewGoDurationValueFromStringMust("30h22m33s"),
+			expectedMatch:   true,
+		},
+		"equal - same duration expressed differently": {
+			currentDuration: timetypes.NewGoDurationValueFromStringMust("3h25m63s"),
+			givenDuration:   timetypes.NewGoDurationValueFromStringMust("12363s"),
+			expectedMatch:   true,
+		},
+		"error - not a Duration value": {
+			currentDuration: timetypes.NewGoDurationValueFromStringMust("56s"),
+			givenDuration:   basetypes.NewStringValue("abcdef"),
+			expectedMatch:   false,
+			expectedDiags: diag.Diagnostics{
+				diag.NewErrorDiagnostic(
+					"Semantic Equality Check Error",
+					"An unexpected value type was received while performing semantic equality checks. "+
+						"Please report this to the provider developers.\n\n"+
+						"Expected Value Type: timetypes.GoDurationType\n"+
```

Comment:
If we update the err msg above, I believe this should be the new message
```suggestion
						"Expected Value Type: timetypes.GoDuration\n"+
```